### PR TITLE
Make swift great again

### PR DESCRIPTION
**Pull request by craigtracey**
Swift was previously using swift-init to manage all of the swift
services. Well it turns out that this leads to some confusion when using
upstart.

To address this, call all of the native binaries in the upstart_services
and reorder when we build the rings. This moves the start/restart
control back into Ansible's hands for now.

Some other small swift updates to make Vagrant great again.

### CODE REVIEW

File: library/swift_disk.py [hunk 1]
```diff
@@ -69,8 +69,12 @@ def main():
     if os.path.exists(part_path) and os.path.ismount(mount_point):
         module.exit_json(changed=False)
 
-    # create partitions
     if dev_path is not None:
+        # create label
+        cmd = ['parted', '--script', dev_path, 'mklabel', 'msdos']
+        rc, out, err = module.run_command(cmd, check_rc=True)    
+
```
Comment (ryshah) [inline library/swift_disk.py]: flotsam bootstrap already creates partition. If we want to recreate one here for vagrant then type should be gpt to handle > 2TB

File: roles/swift-account/tasks/main.yml [hunk 2]
```diff
@@ -1,12 +1,14 @@
 ---
 - name: swift-account service scripts
-  template: src=roles/swift-common/templates/etc/init/swift-service.conf
-            dest=/etc/init/swift-{{ item.service_name }}.conf mode=0644
+  upstart_service: name=swift-{{ item.name }} 
+                   cmd=/usr/local/bin/swift-{{ item.cmd|default(item.name) }}
+                   args=/etc/swift/account-server.conf
+                   user=swift
```
Comment (ryshah) [inline roles/swift-account/tasks/main.yml]: Ah - didn't about this module.. nice change

File: roles/swift-ring/handlers/main.yml [hunk 3]
```diff
@@ -4,22 +4,3 @@
            --config {{ swift_ring.outdir }}/ring_definition.yml --outdir {{ swift_ring.outdir }} bootstrap
   become: True
   become_user: swiftops
-
-- name: setup swift rings - scaleup
```
Comment (ryshah) [inline roles/swift-ring/handlers/main.yml]: Why are we removing these handlers?

File: site.yml [hunk 4]
```diff
@@ -292,6 +292,22 @@
...
 
+- name: install swift-common
+  hosts: swiftnode
+  any_errors_fatal: true
+  roles:
+    - role: swift-common
+      tags: ['openstack', 'swift', 'swift-common', 'data']
+  environment: "{{ env_vars|default({}) }}"
+
+- name: swift bootstrap rings
```
Comment (ryshah) [inline site.yml]: swift-ring role has dependency on swift-common.. so previous block may not be needed?

File: roles/swift-object/tasks/main.yml [hunk 5]
```diff
@@ -65,4 +67,5 @@
     - swift-object
     - swift-object-expirer
     - swift-object-updater
+    - swift-object-auditor
```
Comment (j2sol) [inline roles/swift-object/tasks/main.yml]: Well that's interesting that we were flat out missing this service.

File: site.yml (hunk 4 above)
Comment (j2sol) [inline site.yml]: Agreed, `roles/swift-ring/meta/main.yml` already lists `swift-common` as a requirement.

Comment (j2sol) [inline site.yml]: Although, swift-ring appears to only run on swiftnode-primary, whereas the play above runs swift-common across all the swift nodes.

Without the above play, swifnode_primary would get swift-common, then swift-ring. Then all the swift nodes would get swift-common via the swift-object role in the `swift code and config` role.

We're bootstrapping the rings before doing the code and config play, and the bootstrap uses swiftool, which I believe will contact the other swift nodes and expect some scaffolding to be in place. Perhaps that's why Craig moved the common role to be applied before the ring bootstrap.

**Comment by j2sol** [comment]
There are some open questions here for @craigtracey

File: site.yml (hunk 4 above)
Comment (ryshah) [inline site.yml]: Yeah.. swift-ring expects /etc/swift to copy rings to :+1:

File: library/swift_disk.py (hunk 1 above)
Comment (craigtracey) [inline library/swift_disk.py]: That's fine. This is really meant for vagrant workflow.

File: roles/swift-ring/handlers/main.yml (hunk 3 above)
Comment (craigtracey) [inline roles/swift-ring/handlers/main.yml]: I am not ready to move this in yet. I would like to take a hard look at swifttool in Q3.

File: library/swift_disk.py (hunk 1 above)
Comment (ryshah) [inline library/swift_disk.py]: But it will overwrite the gpt label that was created by flotsam on BM nodes...

Comment (craigtracey) [inline library/swift_disk.py]: "fine" meaning I will address :)

**Comment by ryshah** [comment]
:+1:

**Pull request merged**